File: app/src/main/java/com/bawp/TodoList/Data/TaskDao.java
package com.bawp.TodoList.Data;

import android.view.inspector.PropertyMapper;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.bawp.TodoList.Model.Task;

import java.util.List;

@Dao
public interface TaskDao {

    @Insert
    void insertTask(Task task);

    @Query("DELETE FROM TASK_TABLE")
    void deleteAll();

    @Query("SELECT * FROM TASK_TABLE")
    LiveData<List<Task>> getTasks();

    @Query("SELECT * FROM TASK_TABLE WHERE Task_Id == :id")
    LiveData<Task> get(long id);

    @Update
    void update(Task task);

    @Delete
    void delete(Task task);
}

File: Project_final/data/local/src/main/java/com/escodro/local/dao/TaskDao.kt
package com.escodro.local.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy.REPLACE
import androidx.room.Query
import androidx.room.Update
import com.escodro.local.model.Task

@Dao
interface TaskDao {
    @Insert(onConflict = REPLACE)
    suspend fun insertTask(task: Task)

    @Update
    suspend fun updateTask(task: Task)

    @Delete
    suspend fun deleteTask(task: Task)

    @Query("DELETE FROM task")
    suspend fun cleanTable()

    @Query("SELECT * FROM task where task_due_date is not null")
    suspend fun findAllTasksWithDueDate(): List<Task>

    @Query("SELECT * FROM task WHERE task_id = :taskId")
    suspend fun getTaskById(taskId: Long): Task?
}
